mitiq/folding_cirq.py

```diff
@@ -0,0 +1,61 @@
+import random
+import cirq
+
+def local_folding(circuit, stretch, sampling=False):
+    """Applies the map G -> G G^dag G to ((stretch-1)*len(circuit)//2) to a subset of gates 
+    of the input circuit. For large stretch factors the function is recursively repeated. 
+    Returns a circuit of depth approximately equal to stretch*len(circuit).
+    The stretch factor can be any real number >= 1."""
+
+    if not (stretch >= 1):
+        raise ValueError("The stretch factor must be a real number >= 1.")
+
+    out = cirq.Circuit()
+
+    if stretch <= 3:
+        # select a fraction of subindices
+        d = len(circuit)
+        f = int(d * (stretch - 1) / 2)
```

Comment:
This would be more future proof with more verbose variable names. You never know how far in the future someone will be looking at the code and some additional info in the name is helpful. Perhaps `d --> depth` and `f --> fractional_stretch` ?